src/callback/class-run-sniffer-callback.php

```diff
@@ -262,6 +262,27 @@ final class Run_Sniffer_Callback extends Base_Ajax_Callback {
 	*/
 	const CB_PUBLIC = false;
 
+	/**
+	 * Missing Required Files
+	 *
+	 * @var array
+	 */
+	protected static $missing_files = [];
```

Comment:
I'm just wondering why declaring these variables as static? Wouldn't just declaring them as `protected` work? Then it would be used in the code using `$this->missing_files`, `$this->theme_slug` or `$this->theme_root`, no?

Reply:
Well I was thinking we should probably split this code up to make it a bit easier to follow in the future as more things are added since it's becoming more or less a god class in a way.  My first take was I'll make them static to avoid passing them into each instance, but in looking at it I think we should maybe move these into the sniffer traits, and then create classes per file test, ie the style.css, readme.txt, screenshot.png etc.  Thee theme_root/theme_slug  would probably be common between each (or create a file path var).  The other option maybe would be to allow them to inherit from a parent class.  The instances of each would create the report for the file that gets passed in to the class-run-sniffer-callback.

What do you think?

Reply:
Yeah, the class is becoming a bit big, because the callback is what is running the sniffs.

If we plan to create multiple classes I'd rather try to use dependency injection instead of direct inheritance, to reduce the coupling.

The common methods that are used across the classes can be put in the traits for easier reuse.